src/tm-threads.c

```diff
@@ -2199,14 +2199,24 @@ void TmThreadsSetThreadTimestamp(const int id, const SCTime_t ts)
     int idx = id - 1;
     Thread *t = &thread_store.threads[idx];
     t->pktts = ts;
-    struct timeval systs;
-    gettimeofday(&systs, NULL);
-    t->sys_sec_stamp = (uint32_t)systs.tv_sec;
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+    SCTime_t now = SCTIME_FROM_TIMEVAL(&tv);
+
+    if (t->sys_sec_stamp.secs != 0) {
+        SCTime_t tmpts = SCTIME_ADD_SECS(t->sys_sec_stamp, 3);
+        if (SCTIME_CMP_LT(tmpts, now)) {
+            SCLogNotice("%s: thread slept for %u secs", t->name, (uint32_t)(now.secs - tmpts.secs));
+        }
+    }
+
+    t->sys_sec_stamp = SCTIME_FROM_TIMEVAL(&tv);
     SCMutexUnlock(&thread_store_lock);
 }
 
 bool TmThreadsTimeSubsysIsReady(void)
 {
+    static SCTime_t nullts;
```

Comment:
Should we use `SCTIME_INITIALIZER` just to be sure?

Reply:
Sure